index.js

```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
 		throw new TypeError('Expected a number');
 	}
 
+	opts = opts || {};
+
 	if (ms < 1000) {
+		if (opts.verbose && Math.ceil(ms) === 1.0) {
+			return Math.ceil(ms) + ' millisecond';
+		}
+		if (opts.verbose) {
+			return Math.ceil(ms) + ' milliseconds';
+		}
 		return Math.ceil(ms) + 'ms';
 	}
 
-	opts = opts || {};
-
 	var secDecimalDigits = typeof opts.secDecimalDigits === 'number' ? opts.secDecimalDigits : 1;
 	var ret = [];
 	var parsed = parseMs(ms);
 
-	ret = add(ret, parsed.days, 'd');
-	ret = add(ret, parsed.hours, 'h');
-	ret = add(ret, parsed.minutes, 'm');
+	if (opts.verbose) {
+		ret = add(ret, parsed.days, parsed.days === 1 ? ' day' : ' days');
+		ret = add(ret, parsed.hours, parsed.hours === 1 ? ' hour' : ' hours');
+		ret = add(ret, parsed.minutes, parsed.minutes === 1 ? ' minute' : ' minutes');
+	} else {
+		ret = add(ret, parsed.days, 'd');
+		ret = add(ret, parsed.hours, 'h');
+		ret = add(ret, parsed.minutes, 'm');
+	}
 
 	if (opts.compact) {
-		ret = add(ret, parsed.seconds, 's');
+		if (opts.verbose) {
+			ret = add(ret, parsed.seconds, parsed.seconds === 1 ? ' second' : ' seconds');
+		} else {
+			ret = add(ret, parsed.seconds, 's');
+		}
 		return '~' + ret[0];
 	}
 
-	ret = add(ret, (ms / 1000 % 60).toFixed(secDecimalDigits).replace(/\.0$/, ''), 's');
+	var secondsFixed = (ms / 1000 % 60).toFixed(secDecimalDigits);
+	
+	if (opts.verbose) {
+		ret = add(ret, secondsFixed.replace(/\.0$/, ''), secondsFixed === (1).toFixed(secDecimalDigits) ? ' second' : ' seconds');
```

Comment:
@sindresorhus I am sorry, I don't understand you. What do you mean `vanilla seconds` ?